packages/app-desktop/gui/NoteEditor/utils/resourceHandling.ts

```diff
@@ -125,3 +126,40 @@ export async function handlePasteEvent(event: any) {
 	}
 	return output;
 }
+
+export async function processPastedHtml(html: string) {
+	const allImageUrls: string[] = [];
+	const mappedResources: Record<string, string> = {};
+
+	htmlUtils.replaceImageUrls(html, (src: string) => {
+		allImageUrls.push(src);
+	});
+
+	for (const imageSrc of allImageUrls) {
+		if (!mappedResources[imageSrc]) {
+			try {
+				if (imageSrc.startsWith('file')) {
+					const imageFilePath = imageSrc.substr(7);
+
+					if (imageFilePath.startsWith(Setting.value('resourceDir'))) {
```

Comment:
To compare the paths, please use `path.normalize()` on each of them first. This is in case one of the paths use backslashes and the other forward slashes for example.

Reply:
done.